public/pages/utils/anomalyResultUtils.ts

```diff
@@ -124,24 +124,53 @@ export const prepareDataForChart = (
   if (anomalies.length > MAX_DATA_POINTS) {
     anomalies = sampleMaxAnomalyGrade(anomalies);
   }
-  if (getFloorPlotTime) { // we need to get floor plot time for bar chart
+  if (getFloorPlotTime) {
+    // we need to get floor plot time for bar chart
     anomalies = anomalies.map(anomaly => {
       return {
         ...anomaly,
         plotTime: getFloorPlotTime(anomaly.plotTime),
       };
     });
+    let startTime =
+      anomalies.length > 0
+        ? anomalies[anomalies.length - 1].plotTime
+        : getFloorPlotTime(dateRange.startDate);
+    let endTime =
+      anomalies.length > 0
+        ? anomalies[0].plotTime
+        : getFloorPlotTime(dateRange.endDate);
+
+    while (endTime < dateRange.endDate) {
+      endTime += MIN_IN_MILLI_SECS * interval;
```

Comment:
add comment saying why we need this, such as: "make sure the end of plot time is big enough to cover NOW annotation, which is at dateRange.endDate"

Reply:
sure